src/panels/lovelace/editor/card-editor/hui-card-picker.ts

```diff
@@ -35,77 +39,240 @@ const cards: string[] = [
   "picture-glance",
   "plant-status",
   "sensor",
-  "shopping-list",
   "thermostat",
-  "vertical-stack",
   "weather-forecast",
 ];
 
+const nonPreviewCards: string[] = [
+  "conditional",
+  "entity-filter",
+  "horizontal-stack",
+  "iframe",
+  "vertical-stack",
+  "shopping-list",
+];
+
 @customElement("hui-card-picker")
 export class HuiCardPicker extends LitElement {
-  public hass?: HomeAssistant;
-
+  @property() public hass?: HomeAssistant;
+  public lovelace?: LovelaceConfig;
   public cardPicked?: (cardConf: LovelaceCardConfig) => void;
+  private _unusedEntities?: string[];
+  private _usedEntities?: string[];
+  private _rendered: boolean = false;
```

Comment:
Im thinking maybe I remove Hass as a property. Calculate the entities if undefined in the render?

The issue with has being a propery and the async render card function is that it flashes a bit on update

Reply:
You can better filter the cases you want to update when hass changes, probably only when the language is changed?